hs/src/Reach/Connector/ETH_Solidity.hs

```diff
@@ -994,12 +994,14 @@ solCom = \case
     solIf <$> solArg ca <*> solPLTail t <*> solPLTail f
   DL_LocalSwitch at ov csm -> solSwitch solPLTail at ov csm
   DL_Only {} -> impossible $ "only in CT"
-  DL_ArrayMap _ ans x a i (DLBlock _ _ f r) -> do
-    addMemVars $ [ans, a]
-    let sz = arraySize x
+  DL_ArrayMap _ ans xs as i (DLBlock _ _ f r) -> do
+    addMemVars $ [ans] ++ as
+    let sz = case allEqual $ map arraySize xs of
```

Comment:
You should make a function for this... `arraySizes` and stick it in the same place as `arraySize`